### PR TITLE
Allow fine grained control over the underlying Widgets.

### PR DESCRIPTION
**Pull request by PranayCelllabs**
Exposed **JoystickStick** widget's *color* and *shadowColor* to allow users have fine grained control.
  
Exposed **JoystickBase** widget's  
  - *arrowsColor* to let user decide the colors for the arrows and
  - *withBorderCircle* to allow users decide if they need the circular border around their joystick.
  
These changes allow **fine grained control** for the user and allow them to customize the joystick to **match their theme**.

**Examples:**
Screenshot from 2024-05-30 13-40-41
Screenshot from 2024-05-30 13-38-02
Screenshot from 2024-05-30 13-41-54

### CODE REVIEW

**Comment by PranayCelllabs** [comment]
The example is updated with additional buttons and changes.

Screenshots of new updates:
image
image
image
[Screencast from 06-04-2024 06:34:27 PM.webm](https://github.com/pavelzaichyk/flutter_joystick/assets/99329337/d4d3987a-882e-42aa-953b-d486a52bfee7)

**Comment by pavelzaichyk** [comment]
Hi @PranayCelllabs Thanks for your contribution
There were problems in GitHub Actions that checked the PR. Please rebase your branch onto 'master'; some changes have been added that should fix these problems

File: lib/src/joystick.dart [hunk 1]
```diff
@@ -36,6 +36,8 @@ class Joystick extends StatefulWidget {
   /// Callback, which is called when the stick released.
   final Function? onStickDragEnd;
 
+  final bool includeInitialAnimation;
```
Comment (pavelzaichyk) [inline lib/src/joystick.dart:40]: please add a comment for this new parameter

**Comment by PranayCelllabs** [comment]
Hi Pavel,
I will pull the changes.
Thank you.

File: lib/src/joystick_arrow.dart [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+  });
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    final paint = Paint()
+      ..color = color
+      ..style = PaintingStyle.stroke
+      ..strokeWidth = 3;
+
+    // canvas.drawColor(
```
Comment (pavelzaichyk) [inline lib/src/joystick_arrow.dart]: please remove commented code

File: lib/src/joystick_arrow.dart [hunk 3]
```diff
@@ -0,0 +1,213 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_joystick/src/joystick.dart';
+import 'package:flutter_joystick/src/utils.dart';
+import 'dart:math' as math;
+
+class JoystickArrows extends StatefulWidget {
+  final Color arrowsColor;
+  final JoystickMode mode;
+  final double size;
```
Comment (pavelzaichyk) [inline lib/src/joystick_arrow.dart:9]: could a new parameter be added to disable arrow animation?

File: pubspec.yaml [hunk 4]
```diff
@@ -9,7 +9,7 @@ funding:
   - https://paypal.com/donate/?hosted_button_id=EG9DHRCSB5MY4
 
 environment:
-  sdk: '>=3.1.3 <4.0.0'
+  sdk: ">=3.1.3 <4.0.0"
```
Comment (pavelzaichyk) [inline pubspec.yaml]: please return `'`

File: pubspec.yaml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 name: flutter_joystick
 description: A virtual joystick widget for Flutter applications. Highly flexible and customizable.
-version: 0.0.4
+version: 0.0.5
```
Comment (pavelzaichyk) [inline pubspec.yaml:3]: please list all changes in CHANGELOG.md

**Comment by PranayCelllabs** [comment]
All the suggestions have been implemented. Please, do check and review when possible.

**Comment by pavelzaichyk** [comment]
These changes were published on pub.dev in version 0.1.0. `flutter_joystick: ^0.1.0`

**Pull request merged**